File: packages/flutter_dynamic_icon/android/src/main/java/io/github/tastelessjolt/flutterdynamicicon/Alias.java
package io.github.tastelessjolt.flutterdynamicicon;

public class Alias {
    final String className;
    final String title;
    final int iconResId;

    Alias(String className, String title, int iconResId) {
        this.className = className;
        this.title = title;
        this.iconResId = iconResId;
    }

    @Override
    @SuppressWarnings("NullableProblems")
    public String toString() {
        return title;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        return className.equals(((Alias) o).className);
    }

    @Override
    public int hashCode() {
        return className.hashCode();
    }
}

File: android/app/src/main/kotlin/com/example/dynamic_icon_method_flutter_0604/Alias.kt
package com.example.dynamic_icon_method_flutter_0604

internal class Alias(val className: String, val title: String, val iconResId: Int) {
    val simpleName: String
        get() {
            val index = className.lastIndexOf('.')
            return if (index != -1) className.substring(index + 1) else className
        }

    override fun toString(): String {
        return title
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        return className == (o as Alias).className
    }

    override fun hashCode(): Int {
        return className.hashCode()
    }
}